modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java

```diff
@@ -93,6 +98,33 @@ private Method getCreateArrayOfMethod(Connection delegate)
         return createArrayMethod;
     }
 
+    protected void resolveLobsForArrays(String typeName, Object[] elements) throws SQLException
+    {
+        String nestedTableType = getNestedTableType(typeName);
+
+        if(nestedTableType != null)
+        {
+            typeName = nestedTableType;
+        }
+
+        Map<Integer, String> dataTypes = getDataTypes(typeName);
+
+        if (dataTypes.keySet().isEmpty())
+        {
+            logger.warn("No catalog information was found for the typename '%s'. No lob resolution will be performed", typeName);
+        }
+
+        for (int index : dataTypes.keySet())
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Iterate over the "entrySet" instead of the "keySet". ([squid:S2864](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2864))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7681&issues=AWjt-QE0otIXnmsUG4ve&open=AWjt-QE0otIXnmsUG4ve)

[ISSUE_KEY]: AWjt-QE0otIXnmsUG4ve